macro/src/span.rs

```diff
@@ -0,0 +1,93 @@
+use proc_macro::Span;
+use std::collections::hash_map::DefaultHasher;
+use std::env;
+use std::hash::{Hash, Hasher};
+
+fn seed() -> impl Hash {
+    env::var_os("CONST_RANDOM_SEED")
+}
+
+pub(crate) fn gen_random<T: Random>() -> T {
+    Random::random()
+}
+
+pub(crate) trait Random {
+    fn random() -> Self;
+}
+
+impl Random for u8 {
+    fn random() -> Self {
+        u64::random() as u8
+    }
+}
+
+impl Random for u16 {
+    fn random() -> Self {
+        u64::random() as u16
+    }
+}
+
+impl Random for u32 {
+    fn random() -> Self {
+        u64::random() as u32
+    }
+}
+
+impl Random for u64 {
+    fn random() -> Self {
+        let span = Span::call_site();
+        let mut hasher = DefaultHasher::new();
+        seed().hash(&mut hasher);
+        span.source_file().path().hash(&mut hasher);
+        span.start().line.hash(&mut hasher);
```

Comment:
Is this nightly only?

Reply:
Nightly or dev. See this line:

https://github.com/tkaitchuck/constrandom/blob/d176e118bee4f8cb3597dc13d972564be03fcf7b/macro/src/lib.rs#L1

It won't compile if `--cfg deterministic_based_on_span` is passed to a non-nightly non-dev compiler.

My employer's codebase runs compilers built from source in dev mode.